lib/hal_api/controller/filtering.rb

```diff
@@ -53,6 +53,24 @@ def filters
     @filters ||= parse_filters_param
   end
 
+  def filter_facets
+  end
+
+  def index_collection
+    collection = defined?(super) ? super : HalApi::PagedCollection.new([])
```

Comment:
I kinda thought some of this might end up in or extending paged collection, but I can see the separation of the controller knowing about the facets and setting them, like with other model data. I wonder if the controller or the paged collection should be more aware of the structure of it though...probably not, probably overkill to have mutator and other methods on the collection...maybe a method to add/set a facet?

I guess I am unsure where to put all the filtering functionality. I think your move here to keep it in a concern/mixin may actually be better, as it keeps most of the logic in the same place.

Reply:
I could see doing an optimization like that.  But to start, I wanted to leave that up to the implementing app (augury.prx.org in this case) and see where that takes us.  Rather than locking them into a more controlled data structure.  For the rest ... opened #19.